nemo/collections/asr/parts/multi_head_attention.py

```diff
@@ -306,12 +315,17 @@ def forward(self, x):
             x (torch.Tensor): Its shape is (batch, time, feature_size)
             pos_emb (torch.Tensor): Its shape is (1, time, feature_size)
         """
-        self.extend_pe(x)
+        self.extend_pe(length=x.size(1))
+        if self.pe.dtype != x.dtype or self.pe.device != x.device:
+            self.pe = self.pe.to(device=x.device, dtype=x.dtype)
+
         if self.xscale:
             x = x * self.xscale
 
-        start_pos = (self.pe.size(1) + 1) // 2 - x.size(1)
-        pos_emb = self.pe[:, start_pos:-start_pos]
+        center_pos = self.pe.size(1) // 2
+        start_pos = center_pos - x.size(1) + 1
+        end_pos = center_pos + x.size(1)
+        pos_emb = self.pe[:, start_pos:end_pos]
```

Comment:
Document what these steps are doing, too hard to mentally keep track of tensor shapes here.

Reply:
Done.

Reply:
Cool thanks !